src/apis/impl/UserApiImpl.ts

```diff
@@ -17,27 +18,95 @@ import {PaginatedListMapper} from "../../mappers/paginatedList";
 import {Filter} from "../../filter/Filter";
 import {Connection, ConnectionImpl} from "../../models/Connection";
 import {subscribeToEntityEvents} from "../../utils/rsocket";
+import {Patient} from "../../models/Patient";
+import {UserFilter} from "../../filter";
+import {filteredContactsFromAddresses} from "../../utils/addressUtils";
+import {MessageGatewayApi} from "../MessageGatewayApi";
+import {
+  EmailMessageFactory,
+  SMSMessageFactory
+} from "../../utils/messageGatewayUtils";
 
 export class UserApiImpl implements UserApi {
   private readonly userApi: IccUserApi;
+  private readonly hcpApi: IccHcpartyXApi
   private readonly username: string | undefined;
   private readonly basePath: string;
   private readonly password: string | undefined;
+  private readonly messageGatewayApi: MessageGatewayApi | undefined;
 
-  constructor(api: { cryptoApi: IccCryptoXApi; userApi: IccUserXApi; patientApi: IccPatientXApi; contactApi: IccContactXApi; documentApi: IccDocumentXApi },
+  constructor(api: { healthcarePartyApi: IccHcpartyXApi, cryptoApi: IccCryptoXApi; userApi: IccUserXApi; patientApi: IccPatientXApi; contactApi: IccContactXApi; documentApi: IccDocumentXApi },
+              messageGatewayApi: MessageGatewayApi | undefined,
               basePath: string,
               username: string | undefined,
               password: string | undefined) {
     this.basePath = basePath;
     this.username = username;
     this.password = password;
-    this.userApi = api.userApi
+    this.userApi = api.userApi;
+    this.hcpApi = api.healthcarePartyApi;
+    this.messageGatewayApi = messageGatewayApi;
   }
 
   async checkTokenValidity(userId: string, token: string): Promise<boolean> {
     return this.userApi.checkTokenValidity(userId, token)
   }
 
+  async createAndInviteUser(patient: Patient, messageFactory: SMSMessageFactory | EmailMessageFactory, tokenDuration = 48 * 60 * 60): Promise<User> {
+    // Checks that the Patient has all the required information
+    if (!patient.id) throw new Error("Patient does not have a valid id")
+    if (!patient.firstName) throw new Error("No first name provided in Patient");
```

Comment:
We don't need those two checks : As a patient needs absolutely to have a firstName and a lastName to be created in the kraken, having patient without a firstname and a lastname should not happen. 
However, we can make sure they retrieve a real patient by : 
- checking there is a revision on this patient;
- calling a getPatient request checking the patient exists for real;